warehouse/integrations/azure-synapse/azure-synapse.go

```diff
@@ -210,6 +214,64 @@ func (*AzureSynapse) IsEmpty(_ context.Context, _ model.Warehouse) (empty bool,
 	return
 }
 
+func (as *AzureSynapse) createStagingTable(ctx context.Context, tableName string, columnMap model.TableSchema) (string, error) {
+	stagingTableName := warehouseutils.StagingTableName(
+		provider,
+		tableName,
+		tableNameLimit,
+	)
+
+	cols := warehouseutils.SortColumnKeysFromColumnMap(columnMap)
+	varcharCols, err := as.getStringColumnsWithVariableLength(ctx, tableName)
+	if err != nil {
+		return "", fmt.Errorf("getting varchar columns info: %w", err)
+	}
+
+	columnDefs := lo.Map(cols, func(name string, _ int) string {
+		if dataType, ok := varcharCols[name]; ok {
+			targetType := dataType
+			// use varchar for char and nvarchar for nchar
+			// this is because char and nchar have a max length of 8000 and 4000 respectively.
+			// row size limit is 8060 in Azure Synapse. To fix this, we use varchar and nvarchar with max length
+			switch dataType {
+			case "char":
+				targetType = "varchar"
+			case "nchar":
+				targetType = "nvarchar"
+			}
+			return fmt.Sprintf(`CAST('' AS %[2]s(max)) as %[1]s`, name, targetType)
```

Comment:
@vyeshwanth
1. Values would look like `CAST('' AS nvarchar(max)) as name`?  Does `''` look off? Can you share sample final query for `createStagingTableStmt`?
2. Any impact if we are changing `createStagingTable` for already active connections? Or because its temp table we will use copy, work?

Reply:
```
SELECT TOP 0 CAST('' AS varchar(max)) as id, received_at, CAST('' AS varchar(max)) as t_c_varchar_col, CAST('' AS nvarchar(max)) as t_d_nvarchar_col, CAST('' AS varchar(max)) as t_e_char_col, CAST('' AS nvarchar(max)) as t_f_nchar_col
			INTO
		test_irfxlirjzjpy_1746008650.rudder_staging_load_table_with_variable_character_columns_60b3f72a397e420488ecd4fe4f09f72f
			FROM
		test_irfxlirjzjpy_1746008650.load_table_with_variable_character_columns
```

sample query from test case. 

2. No impact on existing connections. we are staging table with larger character limits using `max`